test/node_smoke_tests/lib/ensure_iterability_es6.js

```diff
@@ -0,0 +1,30 @@
+/* jshint esnext: true */
+
+"use strict";
+
+module.exports = function ensureIterability() {
+	require( "jsdom" ).env( "", function( errors, window ) {
+		if ( errors ) {
+			console.error( errors );
+			process.exit( 1 );
+		}
+
+		var i,
+			ensureJQuery = require( "./ensure_jquery" ),
+			jQuery = require( "../../../dist/jquery.js" )( window ),
+			elem = jQuery( "<div></div><span></span><a></a>" ),
+			result = "";
+
+		ensureJQuery( jQuery );
+
+		for ( i of elem ) {
+			result += i.nodeName;
+		}
+
+		if ( result !== "DIVSPANA" ) {
+			console.error( "for-of doesn't work on jQuery objects; " +
+				"expected: 'DIVSPANA'; got: '" + result + "'" );
```

Comment:
We should probably use `assert`, but I'm fine with a later sweep of `node_smoke_tests` for that.